src/runtime/account_preload.zig

```diff
@@ -0,0 +1,210 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+const runtime = sig.runtime;
+
+const Allocator = std.mem.Allocator;
+
+const Pubkey = sig.core.Pubkey;
+
+const SlotAccountReader = sig.accounts_db.SlotAccountReader;
+
+const AccountSharedData = runtime.AccountSharedData;
+const LoadedTransactionAccounts = runtime.account_loader.LoadedTransactionAccounts;
+const RuntimeTransaction = runtime.transaction_execution.RuntimeTransaction;
+
+/// pubkey -> AccountSharedData for all pubkeys *except* SYSVAR_INSTRUCTIONS_ID,
+/// which is a special case (constructed on a per-transaction basis)
+/// Owns all the accounts it contains.
+pub const AccountMap = std.AutoArrayHashMapUnmanaged(Pubkey, AccountSharedData);
+
+pub fn deinit(map: AccountMap, allocator: Allocator) void {
+    for (map.values()) |account| account.deinit(allocator);
+    var mut = map;
+    mut.deinit(allocator);
+}
+
+/// Initializes a new instance with all the accounts needed for all the
+/// provided transactions.
+pub fn initFromAccountsDb(
+    allocator: Allocator,
+    account_reader: SlotAccountReader,
+    transactions: []const RuntimeTransaction,
+) !std.AutoArrayHashMapUnmanaged(Pubkey, AccountSharedData) {
+    var map: AccountMap = .{};
+    try map.ensureUnusedCapacity(allocator, maxAccounts(transactions));
+    for (transactions) |tx| {
+        try load(&map, allocator, account_reader, &tx.accounts, tx.instructions);
+    }
+    return map;
+}
+
+/// Allocates a sufficiently large account cache to hold every account that
+/// may be needed by these transactions.
+pub fn initSufficientCapacity(allocator: Allocator, transactions: anytype) !AccountMap {
+    var map: AccountMap = .{};
+    try map.ensureUnusedCapacity(allocator, maxAccounts(transactions));
+
+    return map;
+}
+
+/// Counts the maximum number of accounts that may be needed to process these transactions.
+/// This is a naive upper bound that doesn't check for duplicates.
+pub fn maxAccounts(transactions: anytype) usize {
```

Comment:
Do these need to be anytype - what types can this take?

Reply:
I have since removed all the code in this file